domain/tutoringModel/tyche.py

```diff
@@ -0,0 +1,303 @@
+
+        for le in self.learning_elements:
+            # Map learning elements:
+
+            curr_le = le["classification"]
+            if curr_le == "KÜ":
+                classification_learning_element[0] = "kurzuebersicht"
+            elif curr_le == "LZ":
+                classification_learning_element[1] = "lernziele"
+            elif curr_le == "EK":
+                classification_learning_element[2] = "manuskript_ek"
+            elif curr_le == "AB":
+                classification_learning_element[3] = "manuskript_ab"
+            elif curr_le == "BE":
+                classification_learning_element[4] = "manuskript_be"
+            elif curr_le == "RQ":
+                classification_learning_element[5] = "quiz_rq"
+            elif curr_le == "SE":
+                classification_learning_element[6] = "quiz_se"
+            elif curr_le == "ÜB":
+                classification_learning_element[7] = "uebung"
+            elif curr_le == "ZF":
+                classification_learning_element[8] = "zusammenfassung"
+            elif curr_le == "ZL":
+                classification_learning_element[9] = "zusatzmaterial_textuell"
+            elif curr_le == "FO":
+                classification_learning_element[10] = "kollaborativ"
+            elif curr_le == "AN":
+                classification_learning_element[11] = "animation"
+            else:
+                print("LE", curr_le, "is not part of our set!!")
+                raise err.WrongParameterValueError()
+        classification_learning_element = self.set_last_learning_element(
+            classification_learning_element
+        )
+        result_learning_element = []
+        for i in classification_learning_element:
+            if i != "none":
+                result_learning_element.append(i)
+        return result_learning_element
+
+    def RGB_postprocessing(self):
+        curr_le = 0
+        for i in range(len(self.learningpath)):
+            curr_le = self.learningpath[i]
+            if curr_le == "kurzuebersicht":
+                curr_le = "KÜ"
+            elif curr_le == "lernziele":
+                curr_le = "LZ"
+            elif curr_le == "manuskript_ek":
+                curr_le = "EK"
+            elif curr_le == "manuskript_ab":
+                curr_le = "AB"
+            elif curr_le == "manuskript_be":
+                curr_le = "BE"
+            elif curr_le == "quiz_rq":
+                curr_le = "RQ"
+            elif curr_le == "quiz_se":
+                curr_le = "SE"
+            elif curr_le == "uebung":
+                curr_le = "ÜB"
+            elif curr_le == "zusammenfassung":
+                curr_le = "ZF"
+            elif curr_le == "zusatzmaterial_textuell":
+                curr_le = "ZL"
+            elif curr_le == "animation":
+                curr_le = "AN"
+            else:
+                print("Current learning element", curr_le, "is not part of our set!")

```

Comment:
Exception needed or not?

Reply:
Exception is not needed. Not matching learning elements are catched in rgb_preprocessing(). Else-clause is deleted